internal/gatewayapi/resource.go

```diff
@@ -34,21 +34,26 @@ type (
-	TLSRoutes               []*gwapiv1a2.TLSRoute          `json:"tlsRoutes,omitempty" yaml:"tlsRoutes,omitempty"`
-	TCPRoutes               []*gwapiv1a2.TCPRoute          `json:"tcpRoutes,omitempty" yaml:"tcpRoutes,omitempty"`
-	UDPRoutes               []*gwapiv1a2.UDPRoute          `json:"udpRoutes,omitempty" yaml:"udpRoutes,omitempty"`
-	ReferenceGrants         []*gwapiv1b1.ReferenceGrant    `json:"referenceGrants,omitempty" yaml:"referenceGrants,omitempty"`
-	Namespaces              []*v1.Namespace                `json:"namespaces,omitempty" yaml:"namespaces,omitempty"`
-	Services                []*v1.Service                  `json:"services,omitempty" yaml:"services,omitempty"`
-	ServiceImports          []*mcsapi.ServiceImport        `json:"serviceImports,omitempty" yaml:"serviceImports,omitempty"`
-	EndpointSlices          []*discoveryv1.EndpointSlice   `json:"endpointSlices,omitempty" yaml:"endpointSlices,omitempty"`
-	Secrets                 []*v1.Secret                   `json:"secrets,omitempty" yaml:"secrets,omitempty"`
-	ConfigMaps              []*v1.ConfigMap                `json:"configMaps,omitempty" yaml:"configMaps,omitempty"`
-	EnvoyProxy              *egv1a1.EnvoyProxy             `json:"envoyProxy,omitempty" yaml:"envoyProxy,omitempty"`
+	GatewayClass    *gwapiv1.GatewayClass        `json:"gatewayClass,omitempty" yaml:"gatewayClass,omitempty"`
+	Gateways        []*gwapiv1.Gateway           `json:"gateways,omitempty" yaml:"gateways,omitempty"`
+	HTTPRoutes      []*gwapiv1.HTTPRoute         `json:"httpRoutes,omitempty" yaml:"httpRoutes,omitempty"`
+	GRPCRoutes      []*gwapiv1.GRPCRoute         `json:"grpcRoutes,omitempty" yaml:"grpcRoutes,omitempty"`
+	TLSRoutes       []*gwapiv1a2.TLSRoute        `json:"tlsRoutes,omitempty" yaml:"tlsRoutes,omitempty"`
+	TCPRoutes       []*gwapiv1a2.TCPRoute        `json:"tcpRoutes,omitempty" yaml:"tcpRoutes,omitempty"`
+	UDPRoutes       []*gwapiv1a2.UDPRoute        `json:"udpRoutes,omitempty" yaml:"udpRoutes,omitempty"`
+	ReferenceGrants []*gwapiv1b1.ReferenceGrant  `json:"referenceGrants,omitempty" yaml:"referenceGrants,omitempty"`
+	Namespaces      []*v1.Namespace              `json:"namespaces,omitempty" yaml:"namespaces,omitempty"`
+	Services        []*v1.Service                `json:"services,omitempty" yaml:"services,omitempty"`
+	ServiceImports  []*mcsapi.ServiceImport      `json:"serviceImports,omitempty" yaml:"serviceImports,omitempty"`
+	EndpointSlices  []*discoveryv1.EndpointSlice `json:"endpointSlices,omitempty" yaml:"endpointSlices,omitempty"`
+	Secrets         []*v1.Secret                 `json:"secrets,omitempty" yaml:"secrets,omitempty"`
+	ConfigMaps      []*v1.ConfigMap              `json:"configMaps,omitempty" yaml:"configMaps,omitempty"`
+
+	// ClassEnvoyProxy holds EnvoyProxy attached to GatewayClass
+	ClassEnvoyProxy *egv1a1.EnvoyProxy `json:"classEnvoyProxy,omitempty" yaml:"classEnvoyProxy,omitempty"`
```

Comment:
nit: was thinking if we could somehow distinguish EnvoyProxies and ClassEnvoyProxy a bit more, eg. GatewayEnvoyProxies and GCEnvoyProxy

Reply:
```
	// EnvoyProxyForGatewayClass holds EnvoyProxy attached to GatewayClass
	EnvoyProxyForGatewayClass *egv1a1.EnvoyProxy `json:"classEnvoyProxy,omitempty" yaml:"classEnvoyProxy,omitempty"`
	// EnvoyProxiesForGateways holds EnvoyProxiesForGateways attached to Gateways
	EnvoyProxiesForGateways []*egv1a1.EnvoyProxy `json:"EnvoyProxies,omitempty" yaml:"EnvoyProxies,omitempty"`
```
Do we keep classEnvoyProxy and EnvoyProxies as the json name or do we also change them?

Reply:
we should change all